typescript/ws-protocol/src/FoxgloveClient.ts

```diff
@@ -110,6 +112,53 @@ export default class FoxgloveClient {
     return id;
   }
 
+  /**
+   * Advertise a new channel and inform any connected servers.
+   * @returns The id of the new channel
+   */
+  addChannel(channel: Omit<Channel, "id">): ChannelId {
+    const newId = ++this.nextChannelId;
+    const newChannel: Channel = { ...channel, id: newId };
+    this.channelsById.set(newId, newChannel);
+    this.send({ op: "advertise", channels: [newChannel] });
+    return newId;
+  }
+
+  /**
+   * Remove a previously advertised channel and inform any connected servers.
+   */
+  removeChannel(channelId: ChannelId): void {
+    if (!this.channelsById.delete(channelId)) {
+      throw new Error(`Channel ${channelId} does not exist`);
+    }
+    this.send({ op: "unadvertise", channelIds: [channelId] });
+  }
+
+  sendMessageData(subscriptionId: SubscriptionId, timestamp: bigint, payload: BufferSource): void {
```

Comment:
From the discussion on #38, it sounds like JSON would be sufficient for a first pass implementation of publishing. So I would change this to accept an object instead, something like this:
```suggestion
  sendMessage(subscriptionId: SubscriptionId, timestamp: bigint, payload: Record<string, unknown>): void {
```
And we can send a JSON message rather than a binary message on the websocket.